prototests/test_splineBasis.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+from neurostatslib.basis import BSplineBasis, Cyclic_BSplineBasis
+import numpy as np
+
+np.random.seed(10**5)
+
+par_grid = []
+for n_basis in range(1, 10):
+    for order in range(1, n_basis):
+        for window_size in [11,100,1000]:
+            par_grid.append({'n_basis':n_basis, 'order':order, 'window_size':window_size})
+samp = np.random.normal(size=1000)
```

Comment:
I think there should be a way to re-format this so that all the code execution happens inside test functions. Having this executed outside a test function seems a bit messy to me.

Reply:
this is happening inside a test class now